pkg/client/unversioned/jobs.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unversioned
+
+import (
+	"k8s.io/kubernetes/pkg/expapi"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+// JobsNamespacer has methods to work with Job resources in a namespace
+type JobsNamespacer interface {
+	Jobs(namespace string) JobInterface
+}
+
+// JobInterface exposes methods to work on Job resources.
+type JobInterface interface {
+	List(label labels.Selector) (*expapi.JobList, error)
+	Get(name string) (*expapi.Job, error)
+	Create(job *expapi.Job) (*expapi.Job, error)
+	Update(job *expapi.Job) (*expapi.Job, error)
+	Delete(name string) error
+	Watch(label labels.Selector, field fields.Selector, resourceVersion string) (watch.Interface, error)
+}
+
+// jobs implements JobsNamespacer interface
+type jobs struct {
+	r  *ExperimentalClient
+	ns string
+}
+
+// newJobs returns a jobs
+func newJobs(c *ExperimentalClient, namespace string) *jobs {
+	return &jobs{c, namespace}
+}
+
+// List returns a list of jobs that match the label and field selectors.
+func (c *jobs) List(label labels.Selector) (result *expapi.JobList, err error) {
+	result = &expapi.JobList{}
+	err = c.r.Get().Namespace(c.ns).Resource("jobs").LabelsSelectorParam(label).Do().Into(result)
```

Comment:
Follow the pattern of `func (*pods) List(...)` in `pkg/client/unversioned/pods.go`.  That takes both a label selector and a field selector argument.  The fact that there is not a field selector for replicationControllers is a bug.